library/src/lib/icon/icon.component.ts

```diff
@@ -14,22 +14,27 @@ const BASE_ICON_CLASS = 'sap-icon';
 const PREFIX_ICON_CLASS = BASE_ICON_CLASS + '--';
 
 /**
- * The directive that represents an icon. 
+ * The Component that represents an icon.
  *
  * ```html
  * <fd-icon [glyph]="cart-approval" [size]="'l'"></fd-icon>
- * ```
+ // * ```
+ // * Or
+ // * ```html
+ // * <fd-icon [size]="'l'">cart-approval</fd-icon>
+ // * ```
  */
-@Directive({
+@Component({
     // TODO to be discussed
     // tslint:disable-next-line:directive-selector
     selector: 'fd-icon',
+    template: ``,
     host: {
         role: 'presentation'
     }
```

Comment:
Need to set view encapsulation to none since this is a component now

Reply:
What is the motivation for making the icon a component rather than using it as a directive like we do for other widgets that don't need a template definition?  Such as the new `<span fd-badge>`, etc.

Reply:
Eventually I'm planning on having a Material-like experience where you put the icon text inside the icon's content.